src/BaGet/Program.cs

```diff
@@ -51,6 +53,17 @@ await provider
                     // Remove the upload limit from Kestrel. If needed, an upload limit can
                     // be enforced by a reverse proxy server, like IIS.
                     options.Limits.MaxRequestBodySize = null;
+                })
+                .ConfigureAppConfiguration((builderContext, config) =>
+                {
+                    var root = Environment.GetEnvironmentVariable("APPSETTINGS_ROOT");
+                    if (!string.IsNullOrEmpty(root))
+                    {
+                        var env = builderContext.HostingEnvironment;
+                        config.Sources.Clear();
+                        config.AddJsonFile($"{root}/appsettings.json", optional: false, reloadOnChange: true)
```

Comment:
Did you try instead setting `config.SetBasePath(root);`? You may not need to re-add the json file names.

Reply:
If this approach works, can we rename `APPSETTINGS_ROOT` to `BAGET_CONFIG_ROOT`